### PR TITLE
register cr transaction payload

### PR DESCRIPTION
**Pull request by RainFallsSilent**

### CODE REVIEW

File: blockchain/blockvalidator.go [hunk 1]
```diff
@@ -151,6 +151,16 @@ func (b *BlockChain) CheckBlockSanity(block *Block) error {
 			}
 			existingProducerNode[producerNode] = struct{}{}
 		case RegisterCR:
+			crPayload, ok := txn.Payload.(*payload.CRInfo)
+			if !ok {
+				return errors.New("[PowCheckBlockSanity] invalid register producer payload")
+			}
+
+			// Check for duplicate producer in a block
+			if _, exists := existingProducer[crPayload.DID]; exists {
```
Comment (raindust) [inline blockchain/blockvalidator.go]: should be existingCR?

Comment (RainFallsSilent) [reply blockchain/blockvalidator.go]: fixed

File: blockchain/txvalidator_test.go [hunk 2]
```diff
@@ -841,6 +841,142 @@ func (s *txValidatorTestSuite) TestCheckActivateProducerTransaction() {
 		"invalid signature in payload")
 }
 
+func (s *txValidatorTestSuite) TestCheckRegisterCRTransaction() {
```
Comment (raindust) [inline blockchain/txvalidator_test.go]: should add a sanity check too

Comment (RainFallsSilent) [reply blockchain/txvalidator_test.go]: added

File: blockchain/blockvalidator_test.go [hunk 3]
```diff
@@ -4,12 +4,12 @@ import (
 	"bytes"
 	"encoding/hex"
 	"fmt"
+	"github.com/elastos/Elastos.ELA/common/log"
```
Comment (raindust) [inline blockchain/blockvalidator_test.go]: format

Comment (RainFallsSilent) [reply blockchain/blockvalidator_test.go]: fixed

**Pull request merged**